### ISSUE TITLE
The UNC \\ path is lost

### ISSUE THREAD

**Issue by uhtann** [RealRaven2000/FiltaQuilla#79]
The UNC \\\ path is lost after reload Thunderbird.
For example
![t1](https://user-images.githubusercontent.com/75791943/101777827-126d6800-3b04-11eb-833d-2c77c1c199fb.png)
after reload 
![t2](https://user-images.githubusercontent.com/75791943/101778028-5c564e00-3b04-11eb-9b89-601098e4905e.png)

**Comment by RealRaven2000** [comment]
which version of FiltaQuilla, which version of Thunderbird?

**Comment by uhtann** [comment]
TB 78.5.1 64bit FiltaQuilla 3.0pre51

**Comment by RealRaven2000** [comment]
> TB 78.5.1 64bit FiltaQuilla 3.0pre51

great. I will pin the issue for high priority - shouldn't be too hard to fix.

**Comment by RealRaven2000** [comment]
I believe this happens when Thunderbird reads the filters from file:
https://searchfox.org/comm-esr78/source/mailnews/search/src/nsMsgFilterList.cpp#696
Unfortunately I would need a working Visual Studio environment to debug this in C++, which is difficult to set up.

**Comment by ABorrieroItaly** [comment]
Good mornig to all,

this morning I've spent some time on the issue and probably I've solved.

- on my PC, I've closed thunderbird

- I modified the msgFilterRules.dat inserting three slashes (\\\) instead of two before the address 192.168.0.xyz ....; saved and closed

- I've locked the msgFilterRules.dat for any users (SYSTEM, Administrator, Users)  from any writing, leaving only in reading and execution.

- I've restarted and closed Thunderbird several times and ALL WORKS!!!!!


The address in the filter is always \\192.168.0.xyz ...


Thank you


Best regards


A. Borriero

**Comment by ABorrieroItaly** [comment]
Good mornig to all,

this morning I've spent some time on the issue and probably I've solved.

    on my PC, I've closed thunderbird

    I modified the msgFilterRules.dat inserting three slashes (\) instead of two before the address 192.168.0.xyz ....; saved and closed

    I've locked the msgFilterRules.dat for any users (SYSTEM, Administrator, Users) from any writing, leaving only in reading and execution.

    I've restarted and closed Thunderbird several times and ALL WORKS!!!!!

The address in the filter is always \\192.168.0.xyz ... (two slashes)

Thank you

Best regards

A. Borriero

**Comment by RealRaven2000** [comment]
> I modified the msgFilterRules.dat inserting three slashes (\) instead of two before the address 192.168.0.xyz ....; saved and closed
> 
> I've locked the msgFilterRules.dat for any users (SYSTEM, Administrator, Users) from any writing, leaving only in reading and execution.
> 
> I've restarted and closed Thunderbird several times and ALL WORKS!!!!!

what happens if you edit any other filter in the account and then save? I definitely fear Thunderbird will overwrite the three `\\\` if you allow it to save filter rules - the problem is fundamental in the way Thunderbird stores them.

**Comment by ABorrieroItaly** [comment]
... yes, I'm quite sure tha TB will overwrite. 

But - probably - no problem; we can:

-  unlock the file . dat, 
- modify the filter with other items in TB, as we wish
- close TB
- fix manually (TB closed) the three slashes
- lock the .dat file
- open TB

Could be a good solution up to you?

Bye

Andrea

**Comment by ABorrieroItaly** [comment]
... and yes, I agree that it is not a radical and final solution: it is only a by-pass.

Bye

andrea

**Comment by AdamReece-WebBox** [comment]
I did try using `//` instead of `\\` for a UNC path, and `/` instead of `\` for directory separators, which sometimes works on Windows apps. On the plus side it remained in place after Thunderbird restarts, but unfortunately the path isn't resolved so the save/detach action doesn't occur.

**Comment by RealRaven2000** [comment]
that may be a good workaround. I will look into the code whether I can replace that (on Windows) during the file operation. The main problem here is that when you select the path (with Windows Explorer) it will always end up being `\\` . I do not have control on how thunderbird stores this value (this is all built in and hidden to the Add-ons / JavaScript layer)

**Comment by AdamReece-WebBox** [comment]
> The main problem here is that when you select the path (with Windows Explorer) it will always end up being `\\`. I do not have control on how thunderbird stores this value (this is all built in and hidden to the Add-ons / JavaScript layer)

I agree with this entirely, and think it would be acceptable to include a clear warning in the add-on documentation about this whole issue around UNC paths.
